engine/src/main/java/org/hibernate/validator/internal/constraintvalidators/hv/EmailValidator.java

```diff
@@ -85,42 +68,26 @@ public boolean isValid(CharSequence value, ConstraintValidatorContext context) {
 		String localPart = stringValue.substring( 0, splitPosition );
 		String domainPart = stringValue.substring( splitPosition + 1 );
 
-		if ( !matchLocalPart( localPart ) ) {
+		if ( !isValidEmailLocalPart( localPart ) ) {
 			return false;
 		}
 
-		return matchDomain( domainPart );
+		return DomainNameUtil.isValidEmailDomainAddress( domainPart );
 	}
 
-	private boolean matchLocalPart(String localPart) {
+	/**
+	 * Checks validity of emails local part.
+	 *
+	 * @param localPart an email local part to check
+	 *
+	 * @return {@code true} if a given emails local part is valid, {@code false} otherwise
+	 */
+	private boolean isValidEmailLocalPart(String localPart) {
```

Comment:
I'll remove this comment in the final version committed as it's useless (private method and method name and parameter very explicit).